### ISSUE TITLE
time to tag/release 2.20?

### ISSUE THREAD

**Issue by zenczykowski** [radvd-project/radvd#185]
in particular I'm interested in getting pref64 into fedora...

**Comment by agowa338** [comment]
Same. Also I'd like AdvCaptivePortalAPI to get into distros.

**Comment by Neustradamus** [comment]
@robbat2, @stappersg: I think it is needed before Debian 12 freeze.

**Comment by aalmenar** [comment]
I believe a new release is needed since the last one was more than two years ago. Pref64 being one of the lastest additions and some bug fixes.

**Comment by kloczek** [comment]
+1 😋

**Comment by alexhaydock** [comment]
Is there a roadmap to be satisfied before the tagging of a new release?

Seems like there's been a lot of development in the main branch since the last release 2 years ago. Or is it the expectation that downstream distributions should start building from git rather than tagged releases at this point?

**Comment by wuch-g2v** [comment]
+1

**Comment by manonfgoo** [comment]
Id like to get RemoveAdvOnExit into freebsd ports

**Comment by oskar456** [comment]
Ping. Is there anything we could help with getting a released version with all new features?

**Comment by Neustradamus** [comment]
@robbat2: More and more people request a new build, the latest current build is 2.19 (2020-09-20).
I think it is good to do a 2.20, can you do it?

The current changes, 114 commits:
- https://github.com/radvd-project/radvd/compare/v2.19...master

Contributors since 2.19 (in more you) :)
@nomis, @stappersg, @robbat2, @RICCIARDI-Adrien, @haegar, @samboyles1, @steglicd, @jpds, @andrew-sayers, @juhamatk, @manonfgoo, @gaoxingwang, @yas-nyan, @zajdee, @initramfs, @kepstin, @prometheanfire, @thesamesam, @mpontillo, @jetomit

Thanks in advance.

cc @reubenhwk (old maintainer)

**Comment by reubenhwk** [comment]
Is anybody on this?  2.20?  I didn't see a "yes" in the thread.

**Comment by reubenhwk** [comment]
It's been a few years since I've been active on this project, but no, there
was no formal roadmap to be satisfied before cutting a release.  I cut
releases when it had been long enough and there had been enough changes.
It's clearly been long enough, and there have been plenty of changes.

I'll poke around a bit...seems like I left some docs somewhere describing
the process of building the archives and updating the website.

**Comment by reubenhwk** [comment]
Ok if nobody disagrees, I'll push a 2.20 tag, but I think I will skip
updating the old website and uploading packages.

**Comment by stappersg** [comment]
On Sat, Nov 11, 2023 at 07:19:09PM -0800, reubenhwk wrote:
> On Sat, Nov 11, 2023 at 8:57 PM Reuben Hawkins wrote:
> > It's been a few years since I've been active on this project,
> > but no, there was no formal roadmap to be satisfied before cutting
> > a release.  I cut releases when it had been long enough and there
> > had been enough changes.  It's clearly been long enough, and there
> > have been plenty of changes.
> >
> > I'll poke around a bit...seems like I left some docs somewhere
> > describing the process of building the archives and updating the
> > website.

 https://radvd.litech.org/


> Ok if nobody disagrees, I'll push a 2.20 tag,

Not any disagreement, but a very strong wish: Sign the release


> but I think I will skip updating the old website and uploading packages.

Fully agreed.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by manonfgoo** [comment]
Yes Please! Thanks alot for this

From: Geert Stappers ***@***.***>
Reply to: radvd-project/radvd ***@***.***>
Date: Sunday, 12. November 2023 at 09:54
To: radvd-project/radvd ***@***.***>
Cc: manonfgoo ***@***.***>, Mention ***@***.***>
Subject: Re: [radvd-project/radvd] time to tag/release 2.20? (Issue #185)

On Sat, Nov 11, 2023 at 07:19:09PM -0800, reubenhwk wrote:
> On Sat, Nov 11, 2023 at 8:57 PM Reuben Hawkins wrote:
> > It's been a few years since I've been active on this project,
> > but no, there was no formal roadmap to be satisfied before cutting
> > a release. I cut releases when it had been long enough and there
> > had been enough changes. It's clearly been long enough, and there
> > have been plenty of changes.
> >
> > I'll poke around a bit...seems like I left some docs somewhere
> > describing the process of building the archives and updating the
> > website.

https://radvd.litech.org/


> Ok if nobody disagrees, I'll push a 2.20 tag,

Not any disagreement, but a very strong wish: Sign the release


> but I think I will skip updating the old website and uploading packages.

Fully agreed.


Groeten
Geert Stappers
--
Silence is hard to parse

**Comment by robbat2** [comment]
I'll tag a release candidate version after https://github.com/radvd-project/radvd/pull/210 is approved & merged.

I heard a non-public report that newer kernels might have an interaction, so I want to force more testing rather than saying "this is a good public release" - plus we have outstanding issues where we have asked users to test, and they never got back to us.

**Comment by Neustradamus** [comment]
@robbat2 has done a 2.20 RC1 release tag:
- https://github.com/radvd-project/radvd/releases

Can you try and please do a feedback?

**Comment by alexhaydock** [comment]
> @robbat2 has done a 2.20 RC1 release tag:
> 
> * https://github.com/radvd-project/radvd/releases
> 
> Can you try and please do a feedback?

I've flagged this as out-of-date [in Alpine Linux edge](https://pkgs.alpinelinux.org/packages?name=radvd&branch=edge) in the hope it can be brought into edge, or into the testing repo so I can do some testing in my production environment that's based around Alpine Edge.

If not, I'll try and build this manually at some point soon and do some testing, though I'd need to find more time for that.

**Comment by kloczek** [comment]
> @robbat2 has done a 2.20 RC1 release tag:

Please change versioning convention.
Instead `v2.20_rc1` better would be use just `v2.20.rc1`.
More about that versioning you can find on https://discourse.gnome.org/t/new-gnome-versioning-scheme/4235.

**Comment by robbat2** [comment]
@kloczek  why has GNOME moved away from decades of using -/_ for the `rcX` slugs?

Gentoo for example codified common upstream practices to have the underscore separating the core version number. 
https://dev.gentoo.org/~ulm/pms/head/pms.html#section-3.2

**Comment by robbat2** [comment]
Separately, we need to improve the automation to upload proper dist tarballs, since the GitHub release system automatically used the tip of the tree, rather than the files I wanted to upload.

**Comment by stappersg** [comment]
> > @robbat2 has done a 2.20 RC1 release tag:
> 
> Please change versioning convention.
> Instead `v2.20_rc1` better would be use just `v2.20.rc1`.

FWIW, I'm fine  with `v2.20.n` as release candidates and `v2.20.n+1` as actual release. So more _release early, release often._

Less subtle:  _Don't bother about release candidates._

**Comment by kloczek** [comment]
> @kloczek why has GNOME moved away from decades of using -/_ for the `rcX` slugs?

Because:
- `-` is character forbidden by many package manager software to use in version string
- `rc*`, `alpha*`, `beta*`, `0` strings are in alphabetical order 
Both reasons are to make versioning packaging friendly.

**Comment by kloczek** [comment]
> FWIW, I'm fine with `v2.20.n` as release candidates and `v2.20.n+1` as actual release. So more _release early, release often._

That is why better is use sequence `rc*`, `alpha*`, `beta*`, `0` ... `<n>` or subset ot that sequence.

**Comment by alexhaydock** [comment]
My initial testing suggests this builds and is stable on Alpine Linux edge, and I've been able to test and validate that PREF64 extensions are working, as a particular feature which was added between 2.19 and 2.20_RC1

![image](https://github.com/radvd-project/radvd/assets/13152214/23667d4a-fd35-42fa-bd68-38ed0681c950)

I'm pleased to note that this also builds without needing [fix-alpine-plz.patch](https://git.alpinelinux.org/aports/tree/main/radvd/fix-alpine-plz.patch) which Alpine was shipping as a backport of a fix (06689f8c06f44c7e87f7ff1d814428f88375b53f) which was added after 2.19 was tagged.

I'll continue to run this for a while and report back if anything turns up.

**Comment by alexhaydock** [comment]
Where are we at on testing for this one? Is there anything to be satisfied before bringing 2.20 out of RC status?

I notice #227 and #222 are open and were opened after the RC was tagged, but #227 is explicitly marked as needing to wait until after the current release, and I can't see #222 as a showstopper that would block release.

**Comment by cirkku** [comment]
Any luck in getting this released? My impression is that its stable and getting an official release would be necessary for integrating RFC 8781 into VyOS nightly builds.

**Comment by heliosfa** [comment]
I've been doing some testing of this as I am quite keen on the PREF64 support and it all seems happy under Linux (Debian 12), but FreeBSD 14 still needs the patches discussed in #145 from [their ports repo](https://github.com/freebsd/freebsd-ports/tree/main/net/radvd/files) or RADVD doesn't consistently respond to RA solicitations.

Not much is needed and it all seems happy on FreeBSD 14:

```
diff --git a/device-bsd44.c b/device-bsd44.c
index e788c20..a3db461 100644
--- a/device-bsd44.c
+++ b/device-bsd44.c
@@ -127,15 +127,25 @@ ret:

 int setup_allrouters_membership(int sock, struct Interface *iface)
 {
-       return 0;
-
-       /* Quoting:
-       https://github.com/radvd-project/radvd/issues/145#issuecomment-810466476
+       struct ipv6_mreq mreq;
+
+       memset(&mreq, 0, sizeof(mreq));
+       mreq.ipv6mr_interface = iface->props.if_index;
+
+       /* all-routers multicast address */
+       if (inet_pton(AF_INET6, "ff02::2",
+                       &mreq.ipv6mr_multiaddr.s6_addr) != 1) {
+               flog(LOG_ERR, "inet_pton failed");
+               return (-1);
+       }

-       OK, it will be up to FreeBSD guys to show up
-       if they want to upstream their patches.
+       if (setsockopt(sock, IPPROTO_IPV6, IPV6_JOIN_GROUP,
+                       &mreq, sizeof(mreq)) < 0 && errno != EADDRINUSE) {
+               flog(LOG_ERR, "can't join ipv6-allrouters on %s", iface->props.name);
+               return (-1);
+       }

-       */
+       return 0;
 }

 int cleanup_allrouters_membership(int sock, struct Interface *iface)
@@ -184,5 +194,5 @@ int check_ip6_forwarding(void)
 int check_ip6_iface_forwarding(const char *iface)
 {
        dlog(LOG_DEBUG, 4, "checking ipv6 forwarding of interface not supported");
-       return -1;
+       return 1;
 }
diff --git a/interface.c b/interface.c
index 9061435..f47dec5 100644
--- a/interface.c
+++ b/interface.c
@@ -63,6 +63,8 @@ void touch_iface(struct Interface *iface)

 int setup_iface(int sock, struct Interface *iface)
 {
+       int was_ready = iface->state_info.ready;
+
        iface->state_info.changed = 0;
        iface->state_info.ready = 0;

@@ -98,9 +100,12 @@ int setup_iface(int sock, struct Interface *iface)
                dlog(LOG_DEBUG, 5, "no configured AdvRASrcAddress present, skipping send");
                return -6;
        }
+
+       iface->state_info.ready = was_ready;

        /* join the allrouters multicast group so we get the solicitations */
        if (setup_allrouters_membership(sock, iface) < 0) {
+               iface->state_info.ready = 0;
                return -7;
        }
```

**Comment by Neustradamus** [comment]
@robbat2: It is a good time to do it?

cc: @reubenhwk.

**Comment by paravoid** [comment]
Thank you for your work on radvd! We're now past the one year anniversary of 2.20-rc1. Any news here? Any way folks can help?

**Comment by Neustradamus** [comment]
@paravoid: Good catch, 2.20-rc1 (2023-11-17) has more 1 year old.

All, can you look the @joanandk ticket here from last week, maybe good to solve it before?
- https://github.com/radvd-project/radvd/issues/235

Maybe linked to:
- https://github.com/radvd-project/radvd/issues/85
- https://github.com/radvd-project/radvd/issues/91

I think that help is needed.

Thanks in advance.

cc: @reubenhwk, @robbat2, @stappersg.

**Comment by Neustradamus** [comment]
@paravoid: Good catch, 2.20-rc1 (2023-11-17) has more 1 year old.

All, can you look the @joanandk ticket here from last week, maybe good to solve it before?
- https://github.com/radvd-project/radvd/issues/235

Maybe linked to:
- https://github.com/radvd-project/radvd/issues/85 @ihipop, @mhanor, @stu-gott
- https://github.com/radvd-project/radvd/issues/91 @romanrm, @ihipop, @bnutzer, @MarioVerbelen, @ruben-herold, @corvus1, @ipilcher, @stu-gott, @pyk1998

There are too:
- https://github.com/radvd-project/radvd/issues/168 @z1atk0, @gtxaspec
- https://github.com/radvd-project/radvd/issues/228 @xdbob
- https://github.com/radvd-project/radvd/pull/229 @xdbob
- https://github.com/radvd-project/radvd/issues/177 @gaoxingwang
- https://github.com/radvd-project/radvd/pull/178 @gaoxingwang
- https://github.com/radvd-project/radvd/issues/182 @BrainSlayer, @wcbonner, @Mile-Lile, @the-j0k3r, @WenChao1Hou
- https://github.com/radvd-project/radvd/pull/193 @WenChao1Hou, @the-j0k3r, @wcbonner

Help is needed:
- https://github.com/radvd-project/radvd/issues/140
- https://github.com/radvd-project/radvd/issues
- https://github.com/radvd-project/radvd/pulls

Thanks to all in advance.

cc: @reubenhwk, @robbat2, @stappersg, @mpontillo.

**Comment by Neustradamus** [comment]
@robbat2, @stappersg, @reubenhwk: It important to have improvements before Debian 13 freeze.

**Comment by robbat2** [comment]
@Neustradamus  i'm not seeing any announced freeze from Debian Release Management? https://release.debian.org/
Last update there was 2023/12/16

**Comment by melroy89** [comment]
Latest release tag is from 2021, so I'm ready for a new release :)

**Comment by stappersg** [comment]
Help with _"github.com actions"_,  #236, is appreciated.

**Comment by stappersg** [comment]
On Thu, Dec 05, 2024 at 12:00:51PM -0800, Geert Stappers wrote:
> Help with _"github.com actions"_,  #236, is appreciated.

Like help with https://github.com/radvd-project/radvd/pull/237


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by RICCIARDI-Adrien** [comment]
Hi ! Let's have a look !

**Comment by Neustradamus** [comment]
Dear all,

Before 2.20, it will be nice if you can help to solve some Issues/PRs:
- https://github.com/radvd-project/radvd/issues
- https://github.com/radvd-project/radvd/pulls

Recall, some weeks ago, I have done a comment here:
- https://github.com/radvd-project/radvd/issues/185#issuecomment-2494498563

@agowa, @AdSchellevis, @fichtner, @marjohn56, @bmeirellesRJ, @mhanor, @stu-gott, @wcbonner, @the-j0k3r, @WenChao1Hou, @BrainSlayer, @Mile-Lile, @bnutzer, @stu-gott, @MarioVerbelen, @ruben-herold, @ipilcher, @pyk1998, @romanrm, @corvus1, @ihipop

If you know some IPv6 specialists, please come here, there are several tickets, thanks in advance.

**Comment by Neustradamus** [comment]
Dear all,

What do you think about the changes from RFC8319: Support for Adjustable Maximum Router Lifetimes per Link:

Updates to RFC 4861
- https://datatracker.ietf.org/doc/html/rfc8319#section-4

**Comment by zenczykowski** [comment]
https://datatracker.ietf.org/doc/html/rfc8319#section-3

This section makes unrealistic assumptions about multicast packet loss.
Or rather they're reasonable for *wired* networks, but most users' networks are (at least partially) wireless.  True wired only networks, like those in datacenters are often statically configured for reliability (and thus don't run any ipv6 ra daemons).  As such we should definitely do a good job of supporting wireless networks.

Due to a desire for power savings many (most?) battery powered devices (like phones, tablets) have a wifi dtim multiplier > 1 (possibly even as high as 9) which results in multicast/broadcast loss being as high as 90% (though such devices basically just cannot be made to work well with ipv6).

More reasonable devices top out at a dtim multiplier of just m=2, and suffer from 'only' (m-1) / m = 50% multicast (and thus unsolicited RA) loss.

As such K needs to be not 3, but more like 15+.  All lifetimes in the RA should be at *least* K times the MaxRtrAdvInterval.

For power savings reasons you also don't want (any) lifetimes to be below 30 minutes, and ideally closer to 2-3 hours (or more).

As such wrt. https://datatracker.ietf.org/doc/html/rfc8319#section-4:

> In [Section 4.2](https://datatracker.ietf.org/doc/html/rfc8319#section-4.2), inside the paragraph that defines Router Lifetime, change 9000 to 65535 seconds.

sure.

> In [Section 6.2.1](https://datatracker.ietf.org/doc/html/rfc8319#section-6.2.1), inside the paragraph that defines MaxRtrAdvInterval, change 1800 to 65535 seconds.

doesn't hurt, but not at all useful.

> In [Section 6.2.1](https://datatracker.ietf.org/doc/html/rfc8319#section-6.2.1), inside the paragraph that defines AdvDefaultLifetime, change 9000 to 65535 seconds.

sure.

But the real changes should be to the K=3 multipliers, ie.

MinRtrAdvInterval
                     Default: 0.33 * MaxRtrAdvInterval If
^ this should not be 1/3, this should be 1/15 (or even less)

AdvDefaultLifetime
                     Default: 3 * MaxRtrAdvInterval
^ this should not be 3*, this should be 15* (or even more)

**Comment by heliosfa** [comment]
> This section makes unrealistic assumptions about multicast packet loss.
> Or rather they're reasonable for _wired_ networks, but most users' networks are (at least partially) wireless.

I'm not sure it does make unreasonable assumptions at all, given modern networking practices. Multicast on 802.11 networks happens at the minimum data rate of the network, which can potentially be set to sub 10 Mbps. This means it's usually faster to do multicast-to-unicast conversion for WiFi, and many access points do this by default. Whether they do this for IPv6 RAs or not is very much down to implementation...

[RFC9119](https://datatracker.ietf.org/doc/html/rfc9119) has some discussion on the topic.

**Comment by romanrm** [comment]
> Before 2.20, it will be nice if you can help to solve some Issues/PRs

In my view they asked to just tag 2.20 to solidify the current state and have already implemented features get into distros (as there has not been a release in 3 years). It is not feasible to expect to "quickly" solve all the outstanding issues collected over all that time before making the release. Else this can take another 3 years easily.

**Comment by melroy89** [comment]
> It is not feasible to expect to "quickly" solve all the outstanding issues collected over all that time before making the release. Else this can take another 3 years easily.

I actually agree here. Normally, it's better to take the more  "Agile" approach, by releasing more often with smaller releases, and thus smaller iterations. Instead of trying to release one big release every 5 year. This also ensures that you receive timely feedback, which you could include in the new release.

**Comment by Neustradamus** [comment]
I know, I prefer to have several release builds... but currently some are "easy" to fix:
- https://github.com/radvd-project/radvd/pull/193 + https://github.com/radvd-project/radvd/issues/182

Maybe some can help about:
- https://github.com/radvd-project/radvd/issues/168
- https://github.com/radvd-project/radvd/issues/235
- https://github.com/radvd-project/radvd/issues/85 + https://github.com/radvd-project/radvd/issues/91

What do you think about this one:
- https://github.com/radvd-project/radvd/pull/234

**Comment by fichtner** [comment]
FWIW, did a bit of testing in OPNsense/FreeBSD and the current development state seems fine. Found just a small thing in the form of a warning during compile #242

**Comment by stappersg** [comment]
On Mon, Dec 09, 2024 at 04:30:26AM -0800, Franco Fichtner wrote:
> FWIW, did a bit of testing in OPNsense/FreeBSD and the current
> development state seems fine. Found just a small thing in the form of
> a warning during compile #242

Nice.

And yes, I hope to see more "works for me" to this issue
(insteadof opinions, feature requests and "there should be ...").
 

Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by Neustradamus** [comment]
Dear all,

Can you participate on "RDNSS: support more ipv6 addresses" PR?
- https://github.com/radvd-project/radvd/pull/193?

Can you participate on ::/64 problem too?

Thanks in advance.

cc: @crrodriguez, @neocturne, @danrl, @vapier, @GheRivero, @segoon, @moonlinux, @sean797, @ferdinandoterada, @fgont, @nachtgeist, all others.

**Comment by crrodriguez** [comment]
@Neustradamus The last time I wrote code for this project was 10 years ago and while I appreciate and encourge people to ping me on issues they think I can be of help, I'll not be contributing additional effort on this for now.

**Comment by Neustradamus** [comment]
@Leseratte10: Official website updated :)

An announcement will be done on mailing lists.

About https://release-monitoring.org/project/4157/ it is needed to updated with:
- Homepage: https://radvd.litech.org/
- Backend: GitHub
- Version check url: https://github.com/radvd-project/tags

**Comment by robbat2** [comment]
Mailing list is also done; and the website is done
Closing this ticket.

The website lives on github pages now anyway; just has the same hostname to not break historical links.

**Comment by stappersg** [comment]
Uploaded to Debian.

And "unpinned"

![Image](https://github.com/user-attachments/assets/b68fbb68-e3c7-4325-83b8-7ffba0570c91)

**Comment by Neustradamus** [comment]
@zenczykowski and others: Good news, @Leseratte10 has done recently a PR which has been merged about "RFC8319: Support for Adjustable Maximum Router Lifetimes per Link": https://datatracker.ietf.org/doc/html/rfc8319
- https://github.com/radvd-project/radvd/pull/265

It replaces old PRs:
- https://github.com/radvd-project/radvd/pull/190
- https://github.com/radvd-project/radvd/pull/192

It follows my comment here:
- https://github.com/radvd-project/radvd/issues/185#issuecomment-2525423564

It will be in next version.

**Comment by zenczykowski** [comment]
Eh, it doesn't hurt, but it also doesn't actually help.
The problem is the bad defaults (K defaults to 3), not the low maximums.
